game/procurement.py

```diff
@@ -58,14 +55,34 @@ def __init__(
         self.manage_runways = manage_runways
         self.manage_front_line = manage_front_line
         self.manage_aircraft = manage_aircraft
-        self.front_line_budget_share = front_line_budget_share
         self.threat_zones = self.game.threat_zone_for(not self.is_player)
 
+    def calculate_ground_unit_budget_share(self) -> float:
+        armor_investment = 0
+        aircraft_investment = 0
+        for cp in self.owned_points:
+            cp_ground_units = cp.allocated_ground_units(self.game.transfers)
+            armor_investment += cp_ground_units.total_value
+            cp_aircraft = cp.allocated_aircrat(self.game.transfers)
+            aircraft_investment += cp_aircraft.total_value
+
+        total_investment = aircraft_investment + armor_investment
+        # we are in turn 0, or have no units left, both mean we need more aircraft
+        if total_investment is 0:
```

Comment:
`==`, not `is`. `is` is reference equality. It's typically not something that should be used other than checking for `None` or enum equality. This works for small numbers but only because of an implementation detail of cpython that we should not be depending on.